### PR TITLE
[balsa] Parametrize IntegrationTest on HTTP/1 implementation.

### PR DESCRIPTION
**Pull request by bencebeky**
Run IntegrationTest with BalsaParser in addition to http-parser. Disable failing tests for now.

Tracking issue: #21245

Signed-off-by: Bence Béky <bnc@google.com>

Commit Message: [balsa] Parametrize IntegrationTest on HTTP/1 implementation.
Additional Description: Run IntegrationTest with BalsaParser in addition to http-parser. Disable failing tests for now.
Risk Level: low, test-only change
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Review by diannahu — DISMISSED** [review]
Nice parameterization!

File: test/integration/integration_test.cc [hunk 1]
```diff
@@ -55,11 +55,20 @@ void setAllowHttp10WithDefaultHost(
   hcm.mutable_http_protocol_options()->set_default_host_for_http_10("default.com");
 }
 
+std::string testParamToString(
+    const ::testing::TestParamInfo<std::tuple<Network::Address::IpVersion, Http1ParserImpl>>&
```
[the remark this answers is not in the document]
Comment (bencebeky) [reply test/integration/integration_test.cc:59]: Good catch.  I was not aware of that guidance, thank you for digging it up.

File: test/integration/integration_test.cc [hunk 2]
```diff
@@ -55,11 +55,20 @@ void setAllowHttp10WithDefaultHost(
...
+                      TestUtility::http1ParserImplToString(std::get<1>(params.param)));
+}
+
 } // namespace
 
-INSTANTIATE_TEST_SUITE_P(IpVersions, IntegrationTest,
-                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
-                         TestUtility::ipTestParamsToString);
+INSTANTIATE_TEST_SUITE_P(
+    IpVersions, IntegrationTest,
```
[the remark this answers is not in the document]
Comment (bencebeky) [reply test/integration/integration_test.cc:68]: How about IpVersionsAndHttp1Parser as a compromise for brevity?

File: test/integration/integration_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (bencebeky) [reply test/integration/integration_test.cc:59]: On that note, also adding using directives for Combine, Values, and ValuesIn.

**Comment by bencebeky** [comment]
/assign @KBaichoo 

Kevin: PTAL, thanks!

**Comment by KBaichoo** [comment]
can you try merging? I think https://github.com/envoyproxy/envoy/pull/24383 might have renamed some jobs so CI is waiting on them to occur but they won't be unless it's ran again

**Comment by bencebeky** [comment]
> can you try merging? I think #24383 might have renamed some jobs so CI is waiting on them to occur but they won't be unless it's ran again

Thanks for the advice.

**Comment by phlax** [comment]
@bencebeky im confused that this is failing format checks - seems unrelated and merged to `main`

/retest

**Pull request merged**